node/core/pvf/src/worker_common.rs

```diff
@@ -189,6 +189,8 @@ where
 	})
 	.unwrap_err(); // it's never `Ok` because it's `Ok(Never)`
 
+	// TODO: should we call `shutdown` on the stream so that the host is notified?
```

Comment:
I tried looking at the docs and I'm not totally sure, but it seems that `UnixStream` does not do anything to close the connection when it's dropped. If we call `shutdown()` here before the worker exits, then the host can return immediately and doesn't have to wait for the timeout.

Reply:
Well, if I remember correctly how it works (not related to Rust implementation but to Unix domain sockets themselves), calling `shutdown()` on UDS in this case is a good thing to do as a part of socket etiquette but is not strictly necessary when done right before the thread/process exit. When thread exits, the OS cleans up its resources including closing its file descriptors. So the peer will get a notification (zero-size `read()`) as soon as worker thread exits. But feel free to add a `shutdown()` here, just as a common courtesy.